src/Traversals/depthfirst.jl

```diff
@@ -146,3 +146,123 @@ end
 end
 
 @traitfn is_cyclic(g::::(!IsDirected), alg::DepthFirst=DepthFirst()) = ne(g) > 0
+
+
+
+mutable struct BiconnectionState{E <: AbstractEdge} <: AbstractTraversalState
+
+    low::Vector{Int}
+    depth::Vector{Int}
+    stack::Vector{E}
+    components::Vector{Vector{E}}
+    id::Int
+    up::Bool
+    lastnode::Int
+    lastlow::Int
+    have_children::Vector{Bool}
```

Comment:
Can you combine these stacks (`have_children` and `stack`)? `stack::Vector{Tuple{E, Bool}}`

Reply:
Actually, in my tries to improve the performance, i could get rid of have_childern array, but the function remains very slow compared to the original

Reply:
well, let's get rid of any allocations we don't need and then see what else we can do. There's no reason, I don't think, that this should be too much slower than a custom version.

Reply:
the current code has no have_children array